src/components/tree/Download.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC, useLayoutEffect, useMemo, useState } from "react";
+
+import DownloadIcon from "../vectors/Download";
+import { TreeResult } from "../../types/result";
+
+type Props = {
+ treeSections: TreeResult
+}
+
+const Download: FC<Props> = ({ treeSections }) => {
+ const [uri, setUri] = useState<string>("")
+
+ const csv = useMemo(() => {
+  const articles = Object.values(treeSections).flat()
+  let csv: string = Object.keys(articles[0]).sort().join(',')
+
+  csv = csv.replace("leaf,", "")
+  csv = csv.replace("branch,", "")
+  csv = csv.replace("trunk,", "")
+  csv = csv.replace("root,", "")
+
+  for (const article of articles) {
+   csv = csv + "\n" + `${article.authors.join(";").replaceAll(",", "") ?? ""},${article.doi ? article.doi.replaceAll("[", "").replaceAll("]", "").replaceAll(",", ";") : ""},${article.issue ?? ""},${article.journal ? article.journal.replaceAll(",", ";") : ""},${article.keywords.join(";") ?? ""},${article.label ?? ""},${article.page ?? ""},${article.title ? article.title.replaceAll(",", "") : ""},${article.volume ?? ""},${article.year ?? ""}`
+  }
+
+  return csv
```

Comment:
Could you tweak a bit the code so the lines are not long?

You could do:
```
const row = [
  article.authors.join(";").replaceAll(",", "") ?? "",
  article.doi ? article.doi.replaceAll("[", "").replaceAll("]", "").replaceAll(",", ";") : "",
].join(",")
```

Also you can create a function called `articleToRow` where you create functions for each kind of normalization such as `removeComma` and so on, so you can reuse. You could even use regex, like `s.replace(/[\[\]\,]/gm, "")` where `g` means "global" to act as replace all, and `m` means "multiline" in case the description has line breaks for example, so that you can apply the same logic to multiple fields.

Also, to generate the csv, you could install [PapaParse](https://www.papaparse.com/docs#json-to-csv), so that you don't need to format and concat strings.

Note: it's up to you if you want to apply every thing suggested here or just the ones that makes more sense to you.